lib/salestation/web/extractors.rb

```diff
@@ -192,34 +192,38 @@ def self.[](filters:, rack_key:)
         end
 
         def self.extract(filters, request_hash)
-          filters.each_with_object({}) do |filter, extracted_data|
-            case filter
-            when Symbol
-              stringified_key = filter.to_s
-              extracted_data[filter] = request_hash[stringified_key] if request_hash.key?(stringified_key)
-              if extracted_data[filter].is_a?(Array)
-                extracted_data[filter] = extracted_data[filter].map{ |hash|
-                  if hash.is_a?(Hash)
-                    hash.transform_keys(&:to_sym) if hash.is_a?(Hash)
-                  else
-                    hash
-                  end
-                }
-              end
-            when Hash
-              filter.each do |key, nested_filters|
-                stringified_key = key.to_s
-                if request_hash.key?(stringified_key)
-                  value = request_hash.fetch(stringified_key)
-                  extracted_data[key] =
-                    if !value.is_a?(Hash)
-                      value
+          # Filter as a hash is used in some existing implementations that did not expect full 
+          # recursive symbolizing of keys. In this case hash objects at the highest level of object 
+          # are represented as hash of filter keys. This is no longer needed, but we support it 
+          # to avoid a breaking change.
+
+          filters_flat = filters.map{|filter| filter.is_a?(Hash) ? filter.keys : filter}.flatten()
+
+          filters_flat.each_with_object({}) do |filter, extracted_data|
+            symbolized_key = filter 
+            stringified_key = symbolized_key.to_s 
+
+            if request_hash.key?(stringified_key)
```

Comment:
What if the first key is symbolized, but its nested object contains string keys?
`{:key1 => { 'key2' => 'some value'}}`

Reply:
Are values with symbolized keys discarded?

Reply:
Yes, those would indeed be discarded. The same happened in the previous implementation. There are two blocks in the previous implementation - in `when Symbol` block we did `extracted_data[filter] = request_hash[stringified_key] if request_hash.key?(stringified_key)` - I tried it out to be absolutely sure. In case there is no stringified key, the value is discarded, new entry is not created in the extracted_data hash. Same happens in `when Hash` block where we create an entry in extracted_data hash only in condition if `request_hash.key?(stringified_key)`. If I understand it all correctly.

I took it as logical - because I had the impression that the method is supposed to handle only HTTP input and symbolize all the keys. And HTTP params are all initially strings. I checked all the usages (in the 45 projects that I have in my local machine), and I saw that indeed, this is used for HTTP input.

I am not sure if we should be ready that the method may be called with an object that already has some symbolized keys? Since previously it was not the case, then probably not...?